pkg/storaged/vdo-details.jsx

```diff
@@ -0,0 +1,330 @@
+                });
+                return;
+            }
+
+            if (usage.Teardown) {
+                dialog.open({ Title: cockpit.format(_("Please confirm stopping of $0"),
+                                                    vdo.name),
+                              Teardown: usage.Teardown,
+                              Fields: [ ],
+                              Action: {
+                                  Title: _("Stop"),
+                                  action: function () {
+                                      return teardown_active_usage(client, usage)
+                                          .then(function () {
+                                              return vdo.stop();
+                                          });
+                                  }
+                              }
+                });
+            } else {
+                return vdo.stop();
+            }
+        }
+
+        function delete_() {
+            var usage = get_active_usage(client, block? block.path : "/");
+
+            if (usage.Blocking) {
+                dialog.open({ Title: cockpit.format(_("$0 is in active use"), vdo.name),
+                              Fields: [ ],
+                              Blocking: usage.Blocking,
+                });
+                return;
+            }
+
+            function teardown_configs() {
+                if (block) {
+                    return block.Format("empty", { 'tear-down': { t: 'b', v: true } });
+                } else {
+                    return vdo.start()
+                              .then(function () {
+                                  wait_for(client, () => client.slashdevs_block[vdo.dev])
+                                    .then(function (block) {
+                                        return block.Format("empty", { 'tear-down': { t: 'b', v: true } });
+                                    });
+                              })
+                }
+            }
+
+            dialog.open({ Title: cockpit.format(_("Please confirm deletion of $0"),
+                                                vdo.name),
+                          Teardown: usage.Teardown,
+                          Fields: [ ],
+                          Action: {
+                              Title: _("Delete"),
+                              Danger: _("Deleting a VDO device will erase all data on it."),
+                              action: function () {
+                                  return teardown_active_usage(client, usage)
+                                      .then(function () {
+                                          return teardown_configs()
+                                              .then(function () {
+                                                  var location = cockpit.location;
+                                                  return vdo.remove().then(function () {
+                                                      location.go("/");
```

Comment:
Don't we want this to go the storage home page rather than /?

Reply:
`cockpit.location.go` does local navigation and "/" is actually the storage home.  To navigate to the global "/", one needs to use `cockpit.jump`, I think.

Reply:
You are right.